### DISCUSSION TITLE
Does CSS Variables injected dynamically via Javascript ?

### DISCUSSION THREAD

**Discussion by AjayTheWizard** [twbs/bootstrap#37355] [General]
I am currently developing something cool ( also secret because It's may blow your mind :exploding_head: ) in fork of bootstrap.
I saw lot of CSS Variables in CSS Files without any values ( or void ). But In runtime, I saw it's has value. I don't know how. Can anybody help me how CSS Variables got it value during Runtime.

Thanks in Advance

**Comment by julien-deramond** [comment]
Do you have a precise example of a CSS variable without any values but which has a value in runtime so that we could try to provide a precise answer?

**Reply by AjayTheWizard** [reply]
`body` has CSS Variable `--bs-body-text-align` without value

**Reply by julien-deramond** [reply]
OK so you have this rule in __reboot.scss_: `text-align: var(--#{$prefix}body-text-align);` (`$prefix` is set to `bs-` in the project).
This rule is in `body {`.
So, in your browser (or in a CSS file in your project or with inline CSS with `<body style="--bs-body-text-align: end;">` for example) you can do stuff like that `--bs-body-text-align: end;` for example and the body text will be aligned on the right in LTR mode.

Our doc is obviously using Bootstrap so here is the rendering when I change this value dynamically in the browser:

![Screenshot 2022-10-22 at 10 24 48](https://user-images.githubusercontent.com/17381666/197328991-0a5ca663-b459-4fa0-a76a-c58f65b7daf2.png)

More info in [Content > Reboot > CSS variables](https://getbootstrap.com/docs/5.2/content/reboot/#css-variables)
